lib/assert.js

```diff
@@ -886,7 +886,7 @@ assert.ifError = function ifError(err) {
   }
 };
 
-function internalMatch(string, regexp, message, fn) {
+function internalMatch(string, regexp, message, fn, shouldThrows = true) {
```

Comment:
minor nit: `shouldThrow` would sound better IMO

Reply:
Thanks for your review :)